### ISSUE TITLE
v0.64.0 Discussion

### ISSUE THREAD

**Issue by TheSavior** [react-native-community/releases#207]
_This release has not happened yet. This issue is for us to start tracking remaining work and coordinating with our relevant partner teams around the timing of the release_

## Highlighted Changes
- TBD (Someone needs to look through the changelog)
- [Hermes opt-in iOS](https://github.com/facebook/react-native/pull/29914)
- [Inline requires enabled by default](https://github.com/facebook/react-native/pull/29770)
- [new `profile-hermes` CLI command for converting Hermes tracing profile to Chrome format](https://github.com/react-native-community/cli/pull/1246)
- Hermes with Proxy support (will now support react-native-firebase, mobx)
- React 17

## Work Required

- [ ] Changelog
- [ ] [Update documentation](https://github.com/facebook/react-native-website/projects/1) [@simek]
- [ ] Blog post [@grabbou]
- [x] facebook/react-native#29914
- [x] React 17 support (landed)
- [x] facebook/react-native#29882
- [x] facebook/react-native#30177
- [x] facebook/react-native@a7c1c5a
- [x] Migrate warnings in index.js to point to new lean core repos (https://github.com/facebook/react-native/pull/30422)
- [x] The `hermes-engine` pod has no version constraints specified, which should probably happen in the `React-Core.podspec` file and lock it with similar version constraints as the one applied to the Android version in `package.json`. Fixed by https://github.com/facebook/react-native/pull/30432
- [x] The newly introduced codegen invocation step in the iOS Podfile is failing, as the dependencies of the `react-native-codegen` package have not been installed. See output [here](https://gist.github.com/alloy/2d3b3285b76666234be62086f9d4b8b6). In progress PR here https://github.com/facebook/react-native/pull/30449
- [x] Trying to build a newly created Android app fails when checking the dynamic libraries for duplicate classes. See output [here](https://gist.github.com/alloy/a9f2f4102705bad0b2c53407edd2bd89). Fixed by https://github.com/facebook/react-native/pull/30452
- [ ] [_Not_ a RC blocker] facebook/react-native#29984
- [ ] [_Not_ a RC blocker] The new CLI has a minor regression when using a local template dir, now you need to specify the `file://` protocol to resolve that [@grabbou]
- [ ] [_Not_ a RC blocker] There are no instructions in the iOS Podfile template about how to enable Hermes [@grabbou]
- [ ] [_Not_ a RC blocker] Decide whether https://github.com/facebook/react-native/pull/30449 is required for 0.64-stable to be released
- [ ] [_Not_ a RC blocker] When clicking "Open Debugger" on iOS with Hermes and not having Flipper installed, nothing happens. Perhaps a warning / alert or an appropriate log in the CLI would be helpful to instruct those that don't have Flipper installed on the next steps.

## Hopeful Dates

We can never commit to the dates involved in a release as many things are out of our control. However, we are thinking about these rough dates for this release.

- **Cut the release branch:** November 2nd-ish
- **Publish as Stable:** November 9th-ish

**Comment by grabbou** [comment]
Thank you! 🙏 I am about to get a deep dive into the CLI and prepare it for the next release. Will be back here with some features/updates/insights as that happens later this week.

**Comment by rachelnabors** [comment]
I want to thank @simek for volunteering to handle updating the documentation with this release. Updating the API references at the start of the year increased learner satisfaction with the documentation by 70% (woohoo!).But they're very easy to let slide release over release! Until we have found a solid automation solution, it will be up to us to be meticulous and vigilant and keep them up to date with every release :)

**Comment by Simek** [comment]
The stub project for the docs release is here: https://github.com/facebook/react-native-website/projects/1

I have only skimmed over the changeset draft  from this repository #205. I'm waiting for the official changelog in this thread, to start posting issues and implementing the necessary changes.

**Comment by gedeagas** [comment]
I will be happy to help you with the docs @Simek 💖

**Comment by TheSavior** [comment]
I bumped the dates a week, the branch didn't get cut yesterday

**Comment by thymikee** [comment]
RN CLI recently released a minor version with `profile-hermes` command: https://github.com/react-native-community/cli/pull/1246. I think it's worth featuring in the release, even though it's backwards compatible and available in 0.63

**Comment by TheSavior** [comment]
Bumping by another week. React Native EU and the US holiday kept things from moving as we hoped

**Comment by saphal1998** [comment]
@Simek, I'm also dropping a reference to [this PR](https://github.com/facebook/react-native-website/pull/2159) here for reference. I think it can be useful while writing docs for this release.

**Comment by mrousavy** [comment]
> * Hermes with Proxy support (will now support react-native-firebase)

What exactly does that mean?

**Comment by Bardiamist** [comment]
@mrousavy Now Hermes don't support js `Proxy`. So you can't use some packages from `react-native-firebase` which using `Proxy`.

**Comment by mikehardy** [comment]
> > * Hermes with Proxy support (will now support react-native-firebase)
> 
> What exactly does that mean?

@mrousavy / @Bardiamist 

:wave: react-native-firebase maintainer here :-), it is our understanding that our package `@react-native-firebase/database` did not work with Hermes until recently as it required Proxy support. If Hermes supports Proxy now, then the last "not-Hermes-compatible" module for react-native-firebase should work. Success reports welcome of course (we have a few already)

Not sure if this is worth a by-name callout in the release notes or not (not my domain...) but those are the facts as I know them

Cheers

**Comment by Simek** [comment]
Can `PushNotificationIOS` library be removed from the core in this release? Are there any contraindications?

It has been deprecated ~1,5 years ago (https://github.com/facebook/react-native/blame/master/index.js#L386) and it is rarely updated since then. Lately, due to deprecations on the Apple side, it is causing some confusion and issues (https://github.com/facebook/react-native/issues/26199).

CC: @grabbou

**Comment by grabbou** [comment]
Likely to be fixed before we release this version: https://github.com/facebook/react-native/issues/29984

Xcode 12 compatibility.

**Comment by TheSavior** [comment]
@Simek PushNotificationIOS is used in code at FB still. Someone internally will need to refactor that to remove it from core. Probably won't be this release.

**Comment by mikehardy** [comment]
@viniciusfont 

I believe the release workers need the commit hash (of a commit merged to master already) to consider it. Your is:

https://github.com/facebook/react-native/commit/d54113d8c4bcd0e0c7a09acca60819724eb69926

**Comment by kelset** [comment]
Two other things to be considered/done before cutting the 0.64 branch:
* Bump the RN CLI dependency to latest
* Bump Detox dependency / review test suite to 16.x and higher (currently latest is 17.x)

**Comment by rosskhanas** [comment]
Not sure if this is a correct thread to ask, but would it be possible to include https://github.com/facebook/react-native/pull/29728?

**Comment by kelset** [comment]
We should also try to get this merged https://github.com/facebook/react-native/pull/29882 as it is basically a port back to master of a fix that was done locally  in 0.62

**Comment by aleffeh** [comment]
There is some news about Turbo Modules and the new JSI?

**Comment by kelset** [comment]
Now that the main PR still preventing us from cutting the release branch has been merged (Hermes on iOS), let's check in what are we missing still:

PRs we should merge before cutting (or cherry pick afterwards):
* https://github.com/facebook/react-native/pull/30084
* https://github.com/facebook/react-native/pull/29728
* https://github.com/facebook/react-native/pull/29882

Issues that we should solve before 0.64.0:
* https://github.com/facebook/react-native/issues/29984

**Comment by grabbou** [comment]
I have updated the dates in the original issue to better reflect where we are. The work on https://github.com/facebook/react-native/pull/29914 is finally coming to an end (it is blocked by PRs to other repositories). 

I am really optimistic that this one will be ready to go this week.

**Comment by Theofilos-Chamalis** [comment]
Would it be possible to include this PR [facebook/react-native#30177](https://github.com/facebook/react-native/pull/30177) in 0.64.0?

It fixes bundle/release build issues on Android with gradle 6.5+ and android gradle 4.1.0+

**Comment by hosseinmd** [comment]
React 17 is now  landed. I would like to know when that is done to react-native to start an issue in react-native-web too.

**Comment by kuasha420** [comment]
Regarding Proxy support on Hermes, plethora of state management libraries that depend on Proxy support should be mentioned as well, like mobx. 

Question, will the ios version of Hermes have Proxy as well?

**Comment by jeremybarbet** [comment]
@hosseinmd https://reactjs.org/blog/2020/10/20/react-v17.html#react-native

**Comment by hosseinmd** [comment]
I test react 17 in react-native Unexpectedly works.

**Comment by TheSavior** [comment]
> Regarding Proxy support on Hermes, plethora of state management libraries that depend on Proxy support should be mentioned as well, like mobx. 

Thanks, I updated the post to include mobx. Are there others you think should be mentioned that rely on proxy?

> Question, will the ios version of Hermes have Proxy as well? 

I believe so

**Comment by hosseinmd** [comment]
I think react-native need to implement a way to do some jobs in the other thread (because of performance). Something like reanimated 2 is working on it.

**Comment by kuasha420** [comment]
> Thanks, I updated the post to include mobx. Are there others you think should be mentioned that rely on proxy?

easy-peasy comes to mind. I think immer.js also needs proxy. Popular UI framework UI Kitten also uses proxy. 

> I believe so

Super excited to test out 0.64 when it hits RC.

**Comment by terrysahaidak** [comment]
> I think react-native need to implement a way to do some jobs in the other thread (because of performance). Something like reanimated 2 is working on it.

There are quite few of them available right now, for example https://github.com/devfd/react-native-workers

**Comment by hosseinmd** [comment]
> > I think react-native need to implement a way to do some jobs in the other thread (because of performance). Something like reanimated 2 is working on it.
> 
> There are quite few of them available right now, for example https://github.com/devfd/react-native-workers

That is not a good idea, I think the idea of reanimated is so pretty, it works like this

function aFunctionWhichRunningInOtherThread(){
'WorkLet'
 // anything you want to run in another thread
}

**Comment by a-eid** [comment]
@terrysahaidak that repo is no longer maintained.

**Comment by hosseinmd** [comment]
@terrysahaidak you are a reanimated contributor and you know it better than me.

**Comment by kelset** [comment]
@a-eid @hosseinmd @terrysahaidak @kuasha420 I'm hiding your comments since they are OT -- no hard feelings, just trying to keep only relevant comments open to avoid confusion.

---

> React 17 is now landed. I would like to know when that is done to react-native to start an issue in react-native-web too.

@hosseinmd there should be a commit landing in RN master soon to add React 17 support.

---

Let's write once more an overall update on status, since we are getting closer to have the main PR about Hermes in iOS merged:

**Comment by terrysahaidak** [comment]
> @terrysahaidak you are a reanimated contributor and you know it better than me.

@hosseinmd please, open a separate issue in discussion and proposals repository for this discussion - we can continue there.

@kelset please hide this one too, thanks.

**Comment by grabbou** [comment]
https://github.com/facebook/react-native/issues/29984 is fixed by upgrading to CocoaPods 1.10, will be part of my facebook/react-native#29914.

**Comment by TheSavior** [comment]
Re: facebook/react-native#29728 for PlatformColor support for border colors, reading the PR it looks like this only adds support for iOS and not for Android. If that's the case, we probably won't merge it in or prioritize it for 0.64.

**Comment by kuasha420** [comment]
Looks like React 17 got merged in master! [Hooray](https://github.com/facebook/react-native/commit/24bca492c349ab90d40f9444df0f477145a4c311)

**Comment by mikehardy** [comment]
facebook/react-native#30177 mentioned by @Theofilos-Chamalis above in https://github.com/react-native-community/releases/issues/207#issuecomment-714514600 was merged as https://github.com/facebook/react-native/commit/53f55001afbf07494de0df064a92dfdd42f37c98 ✨

**Comment by pavelrotek** [comment]
Could you please merge https://github.com/facebook/react-native/pull/30164 before 0.64. cut?

**Comment by MohamadKh75** [comment]
Since React 17 got merged (https://github.com/facebook/react-native/commit/24bca492c349ab90d40f9444df0f477145a4c311), do we have the new JSX Transform in 0.64? ([here](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html))

**Comment by gaodeng** [comment]
Is it possible to release the 0.64 version without  [Hermes opt-in iOS ](https://github.com/facebook/react-native/pull/29914 ) first?

**Comment by a-eid** [comment]
@gaodeng why, I would love to use the same JavaScript engine on both android & ios for consistency ?

**Comment by kelset** [comment]
@MohamadKh75 I think the answer is yes, since it landed.

@gaodeng that's a good question, I don't think it has been considered so far. I'll try to bring it up as it currently is "the big blocker" and by reading the last few comments on the PR it looks like it's not quite as close as expected to being merged.

**Comment by grabbou** [comment]
Update: the PR is landing and will be merged by tomorrow (most likely). I have updated the original issue and the plan is to cut the branch tomorrow.

@gaodeng in regards to your comment, Hermes will ship as an option in the upcoming release. It's one of the "big things" we have outlined for this particular release. The main reason that PR has been delayed was that, while working on it, we have identified a few (sometimes unrelated) bugs that were fixed at the same time. Fortunately, they are all resolved which will contribute to more stable development workflow anyway, regardless of turning Hermes on/off.

**Comment by a-eid** [comment]
@kelset @grabbou Hermes IOS support was merged 

https://github.com/facebook/react-native/commit/c95ee5ac18e22a32b12bf707bd47b926f09d3d51

**Comment by kuasha420** [comment]
@kelset is the metro bundler and related configs ready to support New JSX Transform in upcoming  RN0.64? I'm not sure.

**Comment by kelset** [comment]
@kuasha420 good question, I've asked internally and while I assume that the answer is yes, I'll update here once I know for sure.

**Comment by kuasha420** [comment]
@kelset Speaking of good questions, here's a bad one. There's so much effort going on about Hermes on IOS but, is that something that Apple would allow in their store? Was there a discussion I missed or does Facebook have that on good authority that Apple's cool with it?

**Comment by dirkpostma** [comment]
@kelset 
Here is a good discussion upon that subject:

https://github.com/facebook/hermes/issues/34

Summary:
- you are allowed to run prepacked code on other interpreters, like Hermes
- it might not be allowed to run dynamic code. like over the air updates, on Hermes.

**Comment by kuasha420** [comment]
@dirkpostma so release bundle should be fine, codepush maybe problematic, Right?

**Comment by zoontek** [comment]
@kuasha420 I guess CodePush shouldn't be an issue either. Hermes consume bytecode compiled AOT (Ahead of Time), so the JS bundle is not included in the app. I don't think CodePush is

**Comment by a-eid** [comment]
@zoontek could you explain what u mean by 

>  Yes, CodePush + Hermes on iOS.

also why was this issue closed ?

**Comment by zoontek** [comment]
@a-eid Sorry, it was a misclick on mobile. Downloading the bytecode bundle consumed by Hermes on iOS OTA (other the air) will violate the Apple App Store rules.

**Comment by kelset** [comment]
> There's so much effort going on about Hermes on IOS but, is this something that Apple would allow in their store? Was there a discussion I missed or does Facebook have it on good authority that Apple's cool with it?

@kuasha420 as pointed out by @dirkpostma, there's a better place to talk about that and it's that issue: https://github.com/facebook/hermes/issues/34

I'm going to mark as OT the comments so to keep it only related to the release process - but please keep the conversation going over there.

Also, AFAIK CodePush is not problematic. About Hermes I haven't dug into it (as I did a few years back for CP) so I can't really answer but looking in that issue it seems to me to understand that there won't be issues.

**Comment by grabbou** [comment]
And, additional, on top of what @kelset already said, this work is preliminary and considered experimental. There will be updates on that matter as soon as we're ready to make such 👌

**Comment by kuasha420** [comment]
Cool cool. Let's get that RC going so us mere mortals can get to testing :) What are we waiting for now?

**Comment by gabrieldonadel** [comment]
Is this release postponed?

**Comment by kelset** [comment]
Hey folks - when working on getting the first RC out we hit a roadblock so we weren't able to publish rc0. The fix should be landing soon, next week we should be able to get it out 🤞

**Comment by wamry** [comment]
what about https://github.com/facebook/react-native/issues/30123, is it going to be included ?

**Comment by kelset** [comment]
@wamry if it landed after you cut the branch, we'll have to cherry pick it. Either way since it's a commit on master it should be portable in 0.64

**Comment by grabbou** [comment]
There are a few outstanding action items needed to cut 0.64 that popped up in the meantime:

- [ ] cherry-pick fixes for the `react-native-codegen` (https://github.com/facebook/react-native/commit/58c80d4f8d82d3867626bb51f3c9fcd917850a4f, https://github.com/facebook/react-native/commit/1bbcbd55ff94087c0a86b87164c223cee302fe4e)
- [ ] fix `publish-npm` step on CircleCI
- [ ] (optional) cherry-pick React sync commits - what are they and are they required? (https://github.com/facebook/react-native/commit/69ade5996f7073d16082af6277eb1388110b171c, https://github.com/facebook/react-native/commit/c7ab78036070b70c014c15029504caa43eaf6925)
- [ ] (optional) bump Metro to latest and [release an appropriate CLI version](https://github.com/react-native-community/cli/pull/1316) to support that as well (https://github.com/facebook/react-native/commit/d8b0997aa936f903becd2c4a64448134853ea37a)

Note: While `publish-npm` remains broken, it shouldn't block us from testing and actually wrapping up the RC. When that step is working, we just need to cherry-pick a fix for it and run ./scripts/bump-oss-version script.

**Comment by grabbou** [comment]
CC: @EliWhite and @rickhanlonii on React sync commits - are they related to previous React sync and should be considered part of 0.64, or something that is work in progress for the future versions?

**Comment by alloy** [comment]
Ok, just spoke with @kelset.

It seems that currently the main external dependency to get v0.64.0 out is having a new release of CLI. The thing I worry about is about _when_ we should be spending our time to test the RC fully from end to end, as that will take a few hours and I don’t want to have to repeat that work in a few days if we already know there will be a new change that needs testing.

Thus:

* @thymikee is it feasible for you to cut a release of the new CLI version  without _any_ v0.64 RC version of RN being available?
  - If it is feasible, then I propose we wait with cutting the RC release until _after_ the new CLI release and test it fully;
  - if it's not feasible, then I propose we cut the first RC but do _not_ spend a lot of time testing it ourselves until you have subsequently released the new CLI version, so that we can test everything we want in the final release.
* Meanwhile we have cherry-picked the following RN commits onto [the 0.64-stable branch](https://github.com/facebook/react-native/commits/0.64-stable), which means other preparation work for the release can happen in parallel (e.g. changelog generation):
  - https://github.com/facebook/react-native/commit/d8b0997aa936f903becd2c4a64448134853ea37a
  - https://github.com/facebook/react-native/commit/d5db2cb5d134e96f591f9a7e988f05d29212e68a
  - https://github.com/facebook/react-native/commit/1bbcbd55ff94087c0a86b87164c223cee302fe4e
  - https://github.com/facebook/react-native/commit/58c80d4f8d82d3867626bb51f3c9fcd917850a4f
  - https://github.com/facebook/react-native/commit/b841ee6fb8df3a91c877e96fe3f653553b619533

**Comment by thymikee** [comment]
> is it feasible for you to cut a release of the new CLI version without any v0.64 RC version of RN being available?

Yes, we can cut a release. Will try to prioritize this today.

**Comment by kelset** [comment]
Perfect!

So, as soon as new CLI version (with metro 0.64 support) is out, a commit on react-native master with the dep bump can be done & merged, and we can cherry pick it to proceed with rc0 (testing and everything).

Aside from that commit (which AFAIK should be the last one strictly needed), I went back into the comments of this issue and found a few more commits that we need to cherry pick (ordered):
* https://github.com/facebook/react-native/commit/00d9deaf6ba26c605694d303bb0cb072fceae5a1
* https://github.com/facebook/react-native/commit/0e9296b95da06789121f052e6cd6d7cac808464c
* https://github.com/facebook/react-native/commit/d85d5d2e1974b463318e4c86da29a5ccdd60a977

**Comment by thymikee** [comment]
CLI v5 alpha is out. PR to core to be cherry-picked: https://github.com/facebook/react-native/pull/30420

**Comment by kelset** [comment]
Thanks for all your work @thymikee#1279 👏 
Looks like the import process for the PR is still ongoing.

I think we might want to do a local version of that commit on branch just to not be blocked by that; also because since the cut there was a commit in particular (this: https://github.com/facebook/react-native/commit/0bbab028d8da0ac7ae6df4626db264127c83f074) which I think will create issues when cherry picking it.

**Comment by thymikee** [comment]
Request to cherry-pick https://github.com/facebook/react-native/commit/a7c1c5aff24671bba609caeb82092a8de3d3b232 by @lukewalczak

**Comment by grabbou** [comment]
All aforementioned commits including @thymike CLI bump and the one requested by @lukewalczak were cherry-picked. @alloy and @kelset are now working on testing the React Native before pushing an RC out.

**Comment by alloy** [comment]
Status update from the trenches ⛑️ 

We hit a few issues while testing:

* [ ] [_Not_ a RC blocker] The new CLI has a minor regression when using a local template dir, now you need to specify the `file://` protocol to resolve that.
* [ ] [_Not_ a RC blocker] Regarding the introduction of Hermes for iOS:
  - The `hermes-engine` pod has no version constraints specified, which should probably happen in the `React-Core.podspec` file and lock it with similar version constraints as the one applied to the Android version in `package.json`.
  - There are no instructions in the iOS Podfile template about how to enable Hermes.
* [ ] [RC **blocker**] The newly introduced codegen invocation step in the iOS Podfile is failing, as the dependencies of the `react-native-codegen` package have not been installed. See output [here](https://gist.github.com/alloy/2d3b3285b76666234be62086f9d4b8b6)

    These dependencies should probably be hoisted to the main RN `package.json` so they get installed when creating the app (?). (Also unsure where the Android side tried to invoke the codegen, so that needs verification too.)
* [ ] [RC **blocker**] Trying to build a newly created Android app fails when checking the dynamic libraries for duplicate classes. See output [here](https://gist.github.com/alloy/a9f2f4102705bad0b2c53407edd2bd89).

**Comment by grabbou** [comment]
Thanks @alloy for the write up.

In regards to:

> The hermes-engine pod has no version constraints specified, which should probably happen in the React-Core.podspec file and lock it with similar version constraints as the one applied to the Android version in package.json.

I would recommend we hardcode it inside `react_native_pods.rb` file, just [like we do for Flipper](https://github.com/facebook/react-native/blob/master/scripts/react_native_pods.rb#L72). I will send a follow up PR to address that, where we can keep the ongoing discussion for more clarity.

> The newly introduced codegen invocation step in the iOS Podfile is failing, as the dependencies of the react-native-codegen package have not been installed. See output here

It would be great to understand the intent of running `react-native-codegen` in the user space, as opposed to doing that before packaging `react-native` for `npm`. I think that the idea was to include user defined packages as well while running the script. While this makes sense long term, maybe we don't need such convenience for `0.64`. 

If we do need to run it in the user land, I see the two following ways to go forward:
- (easier) manually running `yarn install` inside `node_modules/react-native/packages/react-native-codegen`, right before running the build script.
- (harder) releasing a `react-native-codegen` package to `npm` and making it a dependency of `react-native` sounds to me like the easiest solution.

I would argue whether hoisting dependencies is the right way to go. The package was created "for a reason" and I think it is aligned with the general idea of splitting React Native into smaller packages.

**Comment by brunolemos** [comment]
Not sure if this affects 0.64 RC since we don't have it on npm yet, but FYI this is happening with the nightly versions: 

Running `pod install` fails with:

```
[!] CocoaPods could not find compatible versions for pod "React-RCTNetwork":
  In Podfile:
    React-RCTNetwork (from `../../../node_modules/react-native/Libraries/Network`)

None of your spec sources contain a spec satisfying the dependency: `React-RCTNetwork (from `../../../node_modules/react-native/Libraries/Network`)`.

You have either:
 * out-of-date source repos which you can update with `pod repo update` or with `pod install --repo-update`.
 * mistyped the name or version.
 * not added the source repo that hosts the Podspec to your Podfile.
```

PS: Not only `RCTNetwork` but also `RCTSettings`, `RCTImage`, etc

```
macos 11.0.1
node 15.2.1
npm 7.0.8
cocoapods 1.10.0
react-native 0.0.0-c0a299838
yarn 1.22.10 (workspaces)
```

**Comment by alloy** [comment]
@brunolemos Local testing of `pod install` on the current 0.64-stable branch works, so unsure what the issue is that you’re hitting (aside from https://github.com/facebook/react-native/issues/30036, which I know you have worked around locally).

**Comment by grabbou** [comment]
I believe that the fix for codegen landed in master and is ready to be picked: https://github.com/facebook/react-native/commit/e99b8bbb404f8cd1f11b6c7998083be530d7b8a4

This PR https://github.com/facebook/react-native/pull/30449 seems to be doing a different thing (surely not a RC blocker), as it contains both https://github.com/facebook/react-native/commit/e99b8bbb404f8cd1f11b6c7998083be530d7b8a4 + some additional work.

CC: @alloy 

That said, I think all the commits are already in master, can be cherry-picked and should give us a releasable 0.64.0-rc.0.

**Comment by kelset** [comment]
@brunolemos pls check when the latest nightly was released, last I checked there was an issue and latest was over a month old.

**Comment by kelset** [comment]
Looks like we also need this PR: https://github.com/facebook/react-native/pull/30451

And we need to understand what is different in this: https://github.com/facebook/react-native/pull/30449

**Comment by brunolemos** [comment]
> @brunolemos pls check when the latest nightly was released, last I checked there was an issue and latest was over a month old.

@kelset the build was from the previous day, the nightlies seem to be being published normally again

**Comment by kelset** [comment]
@brunolemos ok thanks for clarifying - we'll see if we hit it, there have been a bunch of changes landing over the last few days.

**Comment by grabbou** [comment]
@kelset, the commit you have referenced is already in the updated @alloy response. So yes, I am going to go ahead and cherry-pick it.

Regarding facebook/react-native#30449, I have responded to that a bit earlier as well:

> This PR facebook/react-native#30449 seems to be doing a different thing (surely not a RC blocker), as it contains both facebook/react-native@e99b8bb + some additional work.

**Comment by grabbou** [comment]
Original description of the post updated to keep it structured and organised.

**Comment by grabbou** [comment]
@brunolemos I have run `pod install` and inited a new project on `0.64-stable` and didn't run into an issue that you have referenced. I think it doesn't affect our branch, so I'd go ahead and minimise the comments to make the issue a bit cleaner. 

I would suggest we follow up in the React Native repository, where there's an issue to track nightly builds status.

**Comment by grabbou** [comment]
I can confirm that a fresh new app works just fine on iOS with and without Hermes.

**Comment by alloy** [comment]
I can confirm that a fresh new app works just fine on Android with and without Hermes.

**Comment by grabbou** [comment]
Closing and moving this to #214 as this issue is a bit long already.
